base/abstractarraymath.jl

```diff
@@ -377,7 +377,7 @@ _reperr(s, n, N) = throw(ArgumentError("number of " * s * " repetitions " *
 
     # fill the first inner block
     if all(x -> x == 1, inner)
-        R[axes(A)...] = A
+        R[axes(A)...,:] = A
```

Comment:
Ah, sorry, I wasn't clear!  You'll still need the appropriately sized tuple like you had before, but just a tuple filled with `:`s instead of `OneTo(1)`s.

Reply:
Well...
Looks like both
```julia
R[axes(A)...,:] = A
```
and
```julia
idxs = (axes(A)..., ntuple(n->:, length(shape) - length(inner))...)
R[idxs...] = A
```
throws 
`DimensionMismatch("tried to assign 2×2 array to 2×2×2 destination")`
for `repeat(ones(2,2), 2, 2, 2)`

I think I'll just rollback to the original one.

Reply:
The problem here is that `R` is always a larger array than `A`.
For outer repeat such as `repeat(ones(2,2), 2, 2, 2)`, this line of code simply fills the initial pixels with known values, leaving other pixels still undef. So we can't use `:` in this case.